src/blockchain/compact_filters/mod.rs

```diff
@@ -534,54 +533,62 @@ impl ConfigurableBlockchain for CompactFiltersBlockchain {
 }
 
 /// An error that can occur during sync with a [`CompactFiltersBlockchain`]
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum CompactFiltersError {
     /// A peer sent an invalid or unexpected response
+    #[error("A peer sent an invalid or unexpected response")]
     InvalidResponse,
     /// The headers returned are invalid
+    #[error("Invalid headers")]
     InvalidHeaders,
     /// The compact filter headers returned are invalid
+    #[error("Invalid filter header")]
     InvalidFilterHeader,
     /// The compact filter returned is invalid
+    #[error("Invalid filters")]
     InvalidFilter,
     /// The peer is missing a block in the valid chain
+    #[error("The peer is missing a block in the valid chain")]
     MissingBlock,
     /// Block hash at specified height not found
+    #[error("Block hash not found")]
     BlockHashNotFound,
     /// The data stored in the block filters storage are corrupted
+    #[error("The data stored in the block filters storage are corrupted")]
     DataCorruption,
 
     /// A peer is not connected
+    #[error("A peer is not connected")]
     NotConnected,
     /// A peer took too long to reply to one of our messages
+    #[error("A peer took too long to reply to one of our messages")]
     Timeout,
     /// The peer doesn't advertise the [`BLOOM`](bitcoin::network::constants::ServiceFlags::BLOOM) service flag
+    #[error("Peer doesn't advertise the BLOOM service flag")]
     PeerBloomDisabled,
 
     /// No peers have been specified
+    #[error("No peers have been specified")]
     NoPeers,
 
     /// Internal database error
+    #[error("Internal database error: {0}")]
     Db(rocksdb::Error),
```

Comment:
As we are using `thiserror` the from impls can also be automated using `Variant(#[from] Error_type)` syntax.. 

Same fro other conversions..